lib/collection/src/shards/local_shard/query.rs

```diff
@@ -297,23 +289,74 @@ impl LocalShard {
         search_runtime_handle: &Handle,
         timeout: Option<Duration>,
     ) -> CollectionResult<Vec<ScoredPoint>> {
-        if let Some(rescore) = merge.rescore {
+        let ResultsMerge {
+            limit,
+            order,
+            rescore,
+            filter,
+            score_threshold,
+        } = merge;
+
+        if let Some(rescore_query) = rescore {
             self.rescore(
                 sources,
-                rescore,
-                merge.limit,
+                rescore_query,
+                filter,
+                limit,
+                score_threshold,
                 search_runtime_handle,
                 timeout,
             )
             .await
         } else {
+            let sources = sources.collect_vec();
+
             // no rescore required - just merge, sort and limit
-            let top = sources
+            let filter = filter_with_sources_ids(sources.clone().into_iter(), filter);
```

Comment:
Don't need filter from search results

Reply:
Because there are no pre-fetches in this case, we just return raw query result

Reply:
Agreed. 

Fun detail: after simplifying this bit, it turns out we don't need to specify the order (direction) of results within `ResultsMerge`. So it further simplified the conversion to `PlannedQuery`.